### ISSUE TITLE
Incorrect leading comments

### ISSUE THREAD

**Issue by jrvidal** [eslint/espree#136]
As per eslint/eslint#2368, from the following source:
```
//#!/usr/bin/env node
''
// Foo
''
```

we get an [AST](https://gist.github.com/anonymous/5935938e458ca387e961#file-gistfile1-json-L115) where the shebang comment appears as a `leadingComment` of the second string literal.

**Comment by nzakas** [comment]
Working on this.

**Comment by nzakas** [comment]
This is a lot harder than it seems. Giving up for now. This is completely related to ASI.

**Comment by platinumazure** [comment]
Is this issue easier to resolve with the switch to Acorn?

**Comment by nzakas** [comment]
Good question, it might already be solved, just haven't had the time to check.

**Comment by kaicataldo** [comment]
This looks like it's been fixed by: https://github.com/eslint/espree/issues/256

Running the latest version of Espree on the example code above results in this output:
```js
{
    "type": "Program",
    "start": 0,
    "end": 35,
    "range": [
        22,
        35
    ],
    "body": [
        {
            "type": "ExpressionStatement",
            "start": 22,
            "end": 24,
            "range": [
                22,
                24
            ],
            "expression": {
                "type": "Literal",
                "start": 22,
                "end": 24,
                "range": [
                    22,
                    24
                ],
                "value": "",
                "raw": "''"
            },
            "leadingComments": [
                {
                    "type": "Line",
                    "value": "#!/usr/bin/env node",
                    "start": 0,
                    "end": 21,
                    "range": [
                        0,
                        21
                    ]
                }
            ],
            "trailingComments": [
                {
                    "type": "Line",
                    "value": " Foo",
                    "start": 25,
                    "end": 31,
                    "range": [
                        25,
                        31
                    ]
                }
            ]
        },
        {
            "type": "ExpressionStatement",
            "start": 32,
            "end": 35,
            "range": [
                32,
                35
            ],
            "expression": {
                "type": "Literal",
                "start": 32,
                "end": 34,
                "range": [
                    32,
                    34
                ],
                "value": "",
                "raw": "''"
            },
            "leadingComments": [
                {
                    "type": "Line",
                    "value": " Foo",
                    "start": 25,
                    "end": 31,
                    "range": [
                        25,
                        31
                    ]
                }
            ]
        }
    ],
    "sourceType": "script",
    "comments": [
        {
            "type": "Line",
            "value": "#!/usr/bin/env node",
            "start": 0,
            "end": 21,
            "range": [
                0,
                21
            ]
        },
        {
            "type": "Line",
            "value": " Foo",
            "start": 25,
            "end": 31,
            "range": [
                25,
                31
            ]
        }
    ]
}
```

**Comment by nzakas** [comment]
Thanks, closing.

**Comment by nzakas** [comment]
Oh, we should probably add a test case for this, so reopening.

**Comment by kaicataldo** [comment]
PR with test opened. I used the same comment as in the original post (`//#!/usr/bin/env node`), but am curious why this is a JS comment rather than starting with the shebang like like in the original issue (I see that the shebang doesn't actually parse)
